new: adding files for local php code sniffer using vscode. Resolves #3.

AlanBCole opened the pull request:
First pr for vscode setup. _It works locally on my computer_...

Commit:
adding files for local php code sniffer using vscode

Commit:
Some reorganization

Commit:
reorganizing and working on bash script

Commit:
More reorganizing and some other small changes

Commit:
some fixing up and more reorganizing

cathysnider reviewed (changes requested):
it would be better to append 'vendor' to the .gitignore

Commit:
some renaming to vscode specific names and change how .gitignore file is handled

jwfuller commented on local_code_sniffer/README.md:
```diff
@@ -0,0 +1,45 @@
+5. restart Visual Studio Code.
+
+## Lando setup
+1. In your project's .lando.yml file, make sure that, under `sevices.appserver:`, you have the following:
+    - xdebug: true
+      config:
+        php: .vscode/php.ini
+2. run `lando rebuild` to make the changes in .lando.yml take effect
```
I cannot get this part to work properly
My `.lando.yml` file
```
name: express
recipe: drupal7
config:
  webroot: .
  sevices.appserver:
    xdebug: true
    config:
      php: .vscode/php.ini
```

AlanBCole replied:
Here is a copy of my .lando.yml file (sorry for the length). You can see where the xdebug and config options are set. _Not_ in the root level `config:` but under 
```yaml
services:  
  appserver:
```
- - -

```yaml 
name: fri-339-12-54
recipe: drupal7
config:
  webroot: .
proxy:
  appserver:
    - fri-339-12-54.lndo.site
services:
  appserver:
    build_internal:
      - >-
        curl
        https://github.com/drush-ops/drush/releases/download/8.3.5/drush.phar -L
        -o /tmp/drush.phar && chmod +x /tmp/drush.phar && mv /tmp/drush.phar
        /usr/local/bin/drush && drush --version
    composer: {}
    config:
      php: .vscode/php.ini
    ssl: true
    type: 'php:7.2'
    via: apache
    xdebug: true
    webroot: .
    overrides:
      environment:
        SIMPLETEST_BASE_URL: 'https://appserver'
        SIMPLETEST_DB: 'mysql://drupal7:drupal7@database/drupal7'
  database:
    config: {}
    authentication: mysql_native_password
    type: mysql
    portforward: true
    creds:
      user: drupal7
      password: drupal7
      database: drupal7
tooling:
  composer:
    service: appserver
    cmd: composer --ansi
  db-import <file>:
    service: ':host'
    description: Imports a dump file into a database service
    cmd: /helpers/sql-import.sh
    user: root
    options:
      host:
        description: The database service to use
        default: database
        alias:
          - h
      no-wipe:
        description: Do not destroy the existing database before an import
        boolean: true
  'db-export [file]':
    service: ':host'
    description: Exports database from a database service to a file
    cmd: /helpers/sql-export.sh
    user: root
    options:
      host:
        description: The database service to use
        default: database
        alias:
          - h
      stdout:
        description: Dump database to stdout
  php:
    service: appserver
    cmd: php
  mysql:
    service: ':host'
    description: Drops into a MySQL shell on a database service
    cmd: mysql -uroot
    options:
      host:
        description: The database service to use
        default: database
        alias:
          - h
  drush:
    service: appserver
```

jwfuller commented on local_code_sniffer/README.md:
```diff
@@ -0,0 +1,45 @@
+1. In your project's .lando.yml file, make sure that, under `sevices.appserver:`, you have the following:
+    - xdebug: true
+      config:
+        php: .vscode/php.ini
```
Use code highlighting

jwfuller commented on local_code_sniffer/vscode-setup.sh:
```diff
@@ -0,0 +1,41 @@
+if [[ $rootdir == '' ]]
+then
+    echo ""
+    echo "What is the full path to your project's root directory? >  "
```
```suggestion
    echo "What is the full path to your project's Drupal root directory? >  "
```

jwfuller commented on local_code_sniffer/vscode-setup.sh:
```diff
@@ -0,0 +1,41 @@
+echo -e "vendor\n.vscode\ncomposer.json\ncomposer.lock" >> "$rootdir"/.gitignore
+
+echo ""
+echo "Installing Visual Studio Code extensions ..."
+code --install-extension "felixfbecker.php-debug"
+code --install-extension "felixfbecker.php-intellisense"
+code --install-extension "neilbrayfield.php-docblocker"
+code --install-extension "ikappas.phpcs"
+code --install-extension "persoderlind.vscode-phpcbf"
+code --install-extension "dmitrydorofeev.empty-indent"
```
This gives a number of warnings, are there better versions of these extensions?
```
Installing Visual Studio Code extensions ...
Installing extensions...
Extension 'felixfbecker.php-debug' is already installed.
Installing extensions...
Extension 'felixfbecker.php-intellisense' is already installed.
Installing extensions...
Installing extension 'neilbrayfield.php-docblocker' v2.1.0...
(node:81371) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Extension 'neilbrayfield.php-docblocker' v2.1.0 was successfully installed.
Installing extensions...
Installing extension 'ikappas.phpcs' v1.0.5...
(node:81382) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Extension 'ikappas.phpcs' v1.0.5 was successfully installed.
Installing extensions...
Installing extension 'persoderlind.vscode-phpcbf' v0.0.8...
(node:81393) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Extension 'persoderlind.vscode-phpcbf' v0.0.8 was successfully installed.
Installing extensions...
Installing extension 'dmitrydorofeev.empty-indent' v0.2.0...
(node:81404) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
Extension 'dmitrydorofeev.empty-indent' v0.2.0 was successfully installed.
```

AlanBCole replied:
I have looked at another extension that for both phpcs and phpcbf (called PHP Sniffer & Beautifier valeryanm.vscode-phpsab) which has the same deprecation warnings. I will look into others, as well.

AlanBCole replied:
@jwfuller This seems like it is coming from an npm dependency used in the extension and others that I have checked.

Commit:
Update local_code_sniffer/vscode-setup.sh

Co-authored-by: jwfuller <jwfuller@users.noreply.github.com>

Commit:
changes to README.md to hopefully clear up directions for .lando.yml setup

Commit:
Changing question wording in vscode-setup.sh

Commit:
Workaround for creating breakpoints in PHP files

The most recent release broke the PHP Debug extension created by Felix Becker, which no longer appears to be maintained.

cathysnider reviewed (commented):
Possible improvements to Manual Instructions:

Include update to .gitignore:
echo -e "vendor\n.vscode\ncomposer.json\ncomposer.lock" >> "$rootdir"/.gitignore

CHANGE
1. run `cp composer.json $ROOT`
TO
1. run `cp vscode-dependencies.json $ROOT/composer.json`

CHANGE
`./vendor/bin/phpcs --config-set installed_paths <pathToProjectDirectory>/vendor/drupal/coder/coder_sniffer`
TO
`./vendor/bin/phpcs --config-set installed_paths $ROOT/vendor/drupal/coder/coder_sniffer`

Commit:
fixing up manual set up instructions in README.md file

Commit:
removing phpcs.executablePath from vsc.json settings file. I do not believe this is necessary and it seems to be causing the 'toString of null' error

jwfuller commented:
Worked for me, thanks @AlanBCole!

cathysnider reviewed (changes requested):
In README.md for localcodesniffer folder, (https://github.com/CuBoulder/cu_debug_admin_bundle/blob/feature/716/local_code_sniffer/README.md) the manual instruction "run localcode sniffer = pwd`` 
should be
`localcodesniffer=$PWD`

Commit:
escaping backticks so they display correctly in the rendered README.md file

AlanBCole commented:
@cathysnider I put the change in the README.md. I wanted it to be `` localcodesniffer=`pwd` ``. I just did not use the character escaping correctly so it was not rendering correctly. I changed it so it should be showing up correctly now.

cathysnider commented:
@AlanBCole yes, the back ticks work just as well.
